adal/token_request.py

```diff
@@ -222,15 +222,17 @@ def _get_token_username_password_federated(self, username, password):
 
             wstrust_version = TokenRequest._parse_wstrust_version_from_federation_active_authurl(
                 self._user_realm.federation_active_auth_url)
+            cloud_audience_urn = self._user_realm.cloud_audience_urn
             self._log.debug(
                 'wstrust endpoint version is: %(wstrust_version)s',
                 {"wstrust_version": wstrust_version})
 
             return self._perform_username_password_for_access_token_exchange(
                 self._user_realm.federation_active_auth_url,
-                wstrust_version, username, password)
+                wstrust_version, cloud_audience_urn, username, password)
         else:
             mex_endpoint = self._user_realm.federation_metadata_url
+            cloud_audience_urn = self._user_realm.cloud_audience_urn

```

Comment:
Given that both branches use same data source, I would suggest to move both of them out of the `if ... else ...` branches.

Reply:
Sure, will make this change and then test it for WW flows to verify this behavior change is not affecting anything else. After this, I'll merge this one in.